pkg/river/internal/value/decode.go

```diff
@@ -339,6 +339,17 @@ func tryCapsuleConvert(from Value, into reflect.Value, intoType Type) (ok bool,
 		}
 	}
 
+	// Last attempt: see if the Go types can be converted.
+	if from.Reflect().CanAddr() && from.Reflect().Addr().CanConvert(into.Type()) {
```

Comment:
@erikbaranowski This is primarily where we need the tests, I already covered the expanded definition of capsule types.

(I think the package-level documentation in pkg/river might need to be updated for this too)